### ISSUE TITLE
error reading API key

### ISSUE THREAD

**Issue by NickCassab** [langchain-ai/web-explorer#3]
i tried running this streamlit app but I keep running into a validation error related to finding my openAI API key:

ValidationError: 1 validation error for OpenAIEmbeddings __root__ Did not find openai_api_key, please add an environment variable `OPENAI_API_KEY` which contains it, or pass `openai_api_key` as a named parameter. (type=value_error)

i'm hoping this is just user error, and someone may be able to help me out? thank you

**Comment by jeremyarbid** [comment]
Do you have your api keys in the environment variable?

**Comment by NickCassab** [comment]
yes, i've tried setting my api keys as the environment variable using python, i've tried using a secrets.toml file, and i've tried setting the variables in the script directly. neither method seems to work. Where in the script are the API keys being set and sent to the streamlet app?

![image](https://github.com/langchain-ai/web-explorer/assets/5970253/f7bb5ce5-2c21-4d6c-b63a-3249f2ca3fa0)

**Comment by shashankdeshpande** [comment]
You can configure those as environment variables in script.
For example - 
```python
import os
os.environ["OPENAI_API_KEY"] = "{your-key}"
```

**Comment by jeremyarbid** [comment]
You have a paid OpenAI api, or have quota to use? Are you running the command "streamlit run web_explorer.py" from command terminal? I set my api keys in the secret.toml and it's working for me...

**Comment by NickCassab** [comment]
thank you everyone, i got it working using environment variables, with a .env file
